core/src/main/java/io/questdb/cairo/wal/WalPurgeJob.java

```diff
@@ -476,9 +496,9 @@ private Path setSegmentPath(TableToken tableName, int walId, int segmentId) {
                 .concat(tableName).concat(WalUtils.WAL_NAME_BASE).put(walId).slash().put(segmentId).$();
     }
 
-    private Path segSegmentPendingPath(TableToken tableName, int walId, int segmentId) {
+    private Path setSegmentPendingPath(TableToken tableName, int walId, int segmentId) {
         return path.of(configuration.getRoot())
-                .concat(tableName).concat(WalUtils.WAL_NAME_BASE).put(walId).slash().put(segmentId).put(".pending").$();
+                .concat(tableName).concat(WalUtils.WAL_NAME_BASE).put(walId).slash().put(segmentId).concat(".pending").slash$();
```

Comment:
.slash().put(segmentId) can be replaced with concat

Reply:
Can't: It would pick up the following overload:

```
public Path concat(long pUtf8NameZ) {
```